frontend/src/metabase/dashboard/analytics.ts

```diff
@@ -12,10 +12,25 @@ export const trackAutoApplyFiltersDisabled = (dashboardId: DashboardId) => {
   });
 };
 
-export const trackExportDashboardToPDF = (dashboardId: DashboardId) => {
+type DashboardAccessedVia =
+  | "internal"
+  | "public-link"
+  | "static-embed"
+  | "interactive-iframe-embed"
+  | "sdk-embed";
+
+export const trackExportDashboardToPDF = ({
+  dashboardId,
+  dashboardAccessedVia,
+}: {
+  dashboardId?: DashboardId;
```

Comment:
Hmm.
https://github.com/metabase/metabase/blob/49e6dbdfe3e7924037edb29c9a7282dbe567bc40/frontend/src/metabase-types/api/dashboard.ts#L25

`DashboardId` could either be a string or a number, but we only accept a number in the Snowplow schema.

https://github.com/metabase/metabase/blob/292c6144036a86a53917f43e8e719a0140f92e69/snowplow/iglu-client-embedded/schemas/com.metabase/dashboard/jsonschema/1-1-5#L38-L43

I think we should make the type consistent.

Reply:
I think DashboardId is relatively new and it was probably typed as number before, unfortunately if we the parameter as number now we'll have type errors upstream, I gave it a thought and I think the safest solution is actually by changing the code and not just the type, by only sending the parameter if it's a number, like https://github.com/metabase/metabase/pull/46335/commits/23921ba402923abbfa0a8df4504086ac37688844#diff-7add21813ab25b9039aa3a79f7057e247562a152097d18fa9e6996a1f876066bR34

Do you think this is good enough?

Reply:
That should do. And save us from failing the snowplow getting `string` and fails too.

Also, just to be explicit, I think we should change the type on this line to string as well. What do you think?